lib/data/file/backend.js

```diff
@@ -0,0 +1,110 @@
+import { errors, stringHash } from 'arsenal';
+import crypto from 'crypto';
+import fs from 'fs';
+import config from '../../Config';
+import constants from '../../../constants';
+import Logger from 'werelogs';
+
+const logger = new Logger('FileDataBackend', {
+    logLevel: config.log.logLevel,
+    dumpLevel: config.log.dumpLevel,
+});
+const STORAGE_PATH = config.filePaths.dataPath;
+const FOLDER_HASH = constants.folderHash;
+
+function createLogger(reqUids) {
+    return reqUids ?
+        logger.newRequestLoggerFromSerializedUids(reqUids) :
+        logger.newRequestLogger();
+}
+
+/*
+* Each object/part becomes a file and the files are stored
+* in a directory hash structure
+* under STORAGE_PATH
+*/
+
+
+function getFilePath(key) {
+    const hash = stringHash(key);
+    const folderHashPath = ((hash % FOLDER_HASH) + 1).toString();
+    return `${STORAGE_PATH}/${folderHashPath}/${key}`;
+}
+
+export const backend = {
+    put: function putFile(request, keyContext, reqUids, callback) {
```

Comment:
We add the calculatedHash to it like with the real data backend.  Should be cleaned up in a separate PR.